modules/plugin/mongodb/src/main/java/org/geotools/data/mongodb/MongoSchemaZipFileStore.java

```diff
@@ -0,0 +1,206 @@
+            BufferedReader reader = new BufferedReader(new InputStreamReader(is));
+            // use default implementation
+            return MongoSchemaStore.super.getSimpleFeatureType(reader, name);
+        }
+    }
+
+    @Override
+    public void deleteSchema(Name name) throws IOException {
+        // not creating new zip
+        Map<String, String> zip_properties = new HashMap<>();
+        zip_properties.put("create", "false");
+
+        // get the schema file to delete
+        ZipEntry zipEntry = getSchemaFile(name.getLocalPart());
+        if (zipEntry == null)
+            throw new IOException(
+                    "Unable to find " + name.getLocalPart() + " in " + schemaStoreFile.toURI());
+
+        URI zip_disk = URI.create("jar:" + schemaStoreFile.toURI());
+        // Path path = Paths.get(schemaStoreFile.toURI());
+        try (FileSystem zipfs = FileSystems.newFileSystem(zip_disk, zip_properties)) {
+            // Get the Path inside ZIP File to delete the ZIP Entry
+            Path pathInZipfile = zipfs.getPath(zipEntry.getName());
+            Logger.getGlobal()
+                    .fine(
+                            "About to delete an entry "
+                                    + pathInZipfile.toUri()
+                                    + " from "
+                                    + zip_disk);
+            // Execute Delete
+            Files.delete(pathInZipfile);
+            Logger.getGlobal().info("Deleted entry " + pathInZipfile.toUri() + " from " + zip_disk);
+        }
+    }
+
+    @Override
+    public List<String> typeNames() {
+        List<String> typeNames = new ArrayList<String>();
+        try (ZipFile schemaStoreZipFile = new ZipFile(schemaStoreFile)) {
+
+            Enumeration<? extends ZipEntry> entries = schemaStoreZipFile.entries();
+            while (entries.hasMoreElements()) {
+                ZipEntry entry = entries.nextElement();
+                if (!entry.isDirectory()) {
+                    if (entry.getName().endsWith(MongoSchemaFileStore.SUFFIX_json)) {
+                        String name = entry.getName();
+                        name = name.substring(0, name.indexOf(MongoSchemaFileStore.SUFFIX_json));
+                        typeNames.add(name);
+                    }
+                }
+            }
+        } catch (Exception e) {
+            Logger.getGlobal().log(Level.SEVERE, e.getMessage(), e);
+        }
+
+        return typeNames;
+    }
+
+    private ZipEntry getSchemaFile(String name) {
+        if (!typeNames().contains(name)) return null;
+        try (ZipFile schemaStoreZipFile = new ZipFile(schemaStoreFile)) {
+            return schemaStoreZipFile.getEntry(name + MongoSchemaFileStore.SUFFIX_json);
+        } catch (Exception e) {
+            Logger.getGlobal().log(Level.SEVERE, e.getMessage(), e);
+        }
+        return null;
+    }
+
+    @Override
+    public void close() {
+        // TODO Auto-generated method stub
```

Comment:
Remove the Eclipse autogenerated TODO comment.

Reply:
will do